### PR TITLE
making mojito to work on windows

### PR DESCRIPTION
**Pull request by caridy**
## Goal
- To support windows "partially" (which means, all unit tests and basic features tested, but not functional tests)
- To drop support for `nodejs@0.6` because that version does not support windows
  ## Steps to verify

With this commit very first commit, we get a dummy app working on windows, yay!

```
npm i mojito -g
mojito create app foo
cd foo
mojito create mojit bar
mojito start
```

then navigate to `http://localhost:8666/@bar/index`
## TODO
- [x] Test simple server side app
- [x] Test simple app with `HTMLFrameMojit` deployed to the client
- [x] Test `mojito test app .` in a simple app
- [x] Test rpc calls thru the tunnel (common example)
- [x] Unit tests to pass

### CODE REVIEW

**Comment by drewfish** [comment]
I think some RS addons will need similar tweaks too.

**Comment by caridy** [comment]
Follow-up: windows support is coming: PR #264 #460 #1097 #869 #1063

**Comment by caridy** [comment]
All unit tests are passing except nodejs 0.6, where some of them are failing:

---

Consolidated Test Report Summary

---

Total Number of Executed Tests  :672
Total Number of Passed Tests  : 647
Total Number of Failed Tests  : 13
Total Number of Skipped Tests  : 12

---

List of Failed Test Descriptors

---

Failed Descriptor Path : /home/travis/build/yahoo/mojito/tests/unit/lib/app/addons/rs/rs_test_descriptor.json
Total Number of Failed Tests : 2

Failed Descriptor Path : /home/travis/build/yahoo/mojito/tests/unit/lib/app/autoload/autoload_test_descriptor.json
Total Number of Failed Tests : 3

Failed Descriptor Path : /home/travis/build/yahoo/mojito/tests/unit/lib/app/commands/commands_test_descriptor.json
Total Number of Failed Tests : 1

Failed Descriptor Path : /home/travis/build/yahoo/mojito/tests/unit/lib/lib_test_descriptor.json
Total Number of Failed Tests : 7

File: lib/util.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2013, Yahoo! Inc.  All rights reserved.
+ * Copyrights licensed under the New BSD License.
+ * See the accompanying LICENSE file for terms.
+ */
+
+/*jslint node:true, nomen: true */
+
```
Comment (drewfish) [inline lib/util.js:8]: I think this file might need a `@module` yuidoc section.  (If missing I think its methods get added to the module in the previously processed file :disappointed: .)

**Comment by caridy** [comment]
All unit tests passed in windows for client and server. If travis is ok with this one, we should merge it.

**Comment by caridy** [comment]
`npm test` is not working properly in windows for the mojito repo, apparently related to yahoo/arrow/issues/93

**Comment by drewfish** [comment]
+1

**Pull request merged**